### PR TITLE
mzbuild: break out some *-base Dockerfiles with apt-get and such

### PR DESCRIPTION
**Pull request by danhhz**
This allows us to avoid re-installing the apt things every time we get a CI builder with a cold cache, saving CI time and resources. The time is still dominated by copying huge binaries into the images, but this was easy and should save a few minutes.

We already do this for the persistcli image. Copy the pattern for some other commonly invalidated images.

### Motivation

   * This PR refactors existing code.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: misc/images/prod-base/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# This is a separate mzimage so that we don't have to re-install the apt things
+# every time we get a CI builder with a cold cache.
+
+MZFROM ubuntu-base
+
+RUN apt-get update \
+    && TZ=UTC DEBIAN_FRONTEND=noninteractive apt-get -qy install \
```
[the remark this answers is not in the document]
Comment (benesch) [reply misc/images/prod-base/Dockerfile:16]: It's fine to just add it everywhere. You need it for the transitive deps of some of these packages. Can never remember which, and it's hard to trace through the transitive graphs anyway.

File: misc/images/prod-base/mzbuild.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+name: prod-base
```
[the remark this answers is not in the document]
Comment (benesch) [reply misc/images/prod-base/mzbuild.yml:10]: Seems good!

File: src/ore/src/lib.rs [hunk 3]
```diff
@@ -87,6 +87,8 @@
 //! Modules are included in this crate when they are broadly useful but too
 //! small to warrant their own crate.
 
+// WIP testing
```
Comment (benesch) [inline src/ore/src/lib.rs:90]: (don't forget to remove before merge)

File: misc/images/prod-base/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        ca-certificates \
+        curl \
+        tini \
+        ssh \
+    && groupadd --system --gid=999 materialize \
+    && useradd --system --gid=999 --uid=999 --create-home materialize \
+    && mkdir /scratch \
```
[the remark this answers is not in the document]
Comment (benesch) [reply misc/images/prod-base/Dockerfile:23]: I don't mind having them here! At least not while they apply equally well to all of the images that are derived from this one. (I guess `balancerd` doesn't care about `/scratch`, but whatever; it seems like a general purpose enough concept that you could imagine `balancerd` one day wanting.)

File: src/ore/src/lib.rs (hunk 3 above)
Comment (danhhz) [reply src/ore/src/lib.rs:90]: Done!

**Review by danhhz — COMMENTED** [review]
Marking this ready for review, but someone will have to make the `prod-base`, `materialized-base`, and `testdrive-base` thingys in docker hub before CI will pass.

**Review by benesch — APPROVED** [review]

**Pull request merged**